### ISSUE TITLE
False positives for at-rules within selectors lists in indentation

### ISSUE THREAD

**Issue by AndyOGo** [stylelint/stylelint#3168]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A feature where using SCSS's `@at-root` selector syntax breaks my `indentation` rule.
So I would like to be able to ignore custom keywords, like `@at-root`

> Which rule, if any, is this issue related to?

`indentation`

> What SCSS is needed to reproduce this issue?

````scss
.foo {
  text-align: center;

  @at-root .foo--left,
  .text-left & {
    text-align: left;
  }
}
````

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "extends": "stylelint-config-standard"
}
```

> Which version of stylelint are you using?

`8.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

NPM script with `"test-stylelint": "stylelint 'src/**/*.scss'",`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS `@at-root` selector syntax.

> What did you expect to happen?

No errors to be thrown.

> What actually happened (e.g. what warnings or errors you are getting)?

The following errors were flagged:

```shell
src/components/m-header-navigation/index.scss
 59:3  ✖  Expected indentation of 4 spaces   indentation
```

**Note**
I also created an issue at `stylelint-scss`, but it wouldn't make much sense to basically copy the most of the `indentation` rule just for scss.
https://github.com/kristerkari/stylelint-scss/issues/206

**Comment by jeddy3** [comment]
@AndyOGo Thanks for the report and for using the template.

That is some bizarre syntax... an at-rule within a selector list.

Please consider investigating further as I don't think many of the core team use non-standard syntax these days, and so we rely on external contributors to fix bugs related to them. Perhaps start by looking into [postcss-scss](https://github.com/postcss/postcss-scss)'s compatibility with such syntax.

**Comment by AndyOGo** [comment]
@jeddy3 Thanks for your answer.

Indeed, it's very SCSS-specific feature, document at whole here:
http://sass-lang.com/documentation/file.SASS_REFERENCE.html#at-root

I basically has two applications:
- with and without parameter
- used inline for a single selector, or containing a block of selectors

I would favor a generic solution, e.g. by ignoring based on a list of keywords.

**Comment by jeddy3** [comment]
> I would favor a generic solution, e.g. by ignoring based on a list of keywords, like `ignoreAtRules: ["at-root"]`

PostCSS parses the construct [as so](https://astexplorer.net/#/gist/86338c37f4b17d04282dcb68dbe8a7fe/3ba22eb09e021a52541550b3a6d7279173538629), where `.foo--left, .text-left &` is considered the _params_ of `@at-root`. By default, the `indentation` rule expects params on a new line to be indented.

So, I have some reservations that it might be confusing for users to use an option like `ignoreAtRules` because it requires an understanding of how PostCSS parses the construct.

The `indentation` rule is a bit of an oddity... all the other rules check that something conforms to standard syntax before checking it, whereas `indentation` specifically deals with non-standard features and/or syntax. For example, there is an at-rule called `@apply` that is parsed in a similar way to `at-root` and the `indentation` rules [specifically deals with it](https://github.com/stylelint/stylelint/blob/master/lib/rules/indentation/index.js#L209-L214).

I think specifically dealing with `at-root` in this same way is fine for now.

**Comment by jeddy3** [comment]
@AndyOGo If you'd like to contribute a fix, you'll find [instructions in the developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by AndyOGo** [comment]
@jeddy3 Sorry, atm I'm super busy with customers.

**Comment by AndyOGo** [comment]
@jeddy3 That was a very small change. Just submitted a PR. Though prettier won't let me commit, so I forced - could not autofix and I can't tell what's wrong with my formatting 🤔
